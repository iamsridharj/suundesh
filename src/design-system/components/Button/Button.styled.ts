import styled from 'styled-components'
import { ButtonProps } from './Button.types'

export const StyledButton = styled.button<{
  $iconPosition: ButtonProps['iconPosition']
}>`
  padding: 8px 16px;
  border-radius: 5px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  border-style: solid;
  border-width: 2px;
  border-radius: 5rem;
  transition: background-color .2s ease-out;
  flex-direction: ${({ $iconPosition }) =>
    $iconPosition === 'left' ? 'row-reverse' : 'row'};

  color: ${({ theme }) => {
    return theme.colors.white
  }};
  border: 1px solid ${({ theme }) => {
      return theme.colors.interactiveAccent
    }};
  background: ${({ theme }) => {
    return theme.colors.interactiveAccent
  }};

  &:hover {
    background: ${({ theme }) => {
    return theme.colors.secondary
  }};
  border: 1px solid ${({ theme }) => {
    return theme.colors.secondary
  }};
  }
`
