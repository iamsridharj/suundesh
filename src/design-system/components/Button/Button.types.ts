import { ReactNode } from 'react'

export interface ButtonProps {
  icon?: ReactNode
  variant?: 'default' | 'danger' | 'success'
  iconPosition?: 'left' | 'right'
  ariaLabel?: string
  onClick?: () => void
}
