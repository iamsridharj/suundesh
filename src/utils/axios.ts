import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';

const baseURL = "https://dev-api.konfhub.com";

const axiosWrapper: AxiosInstance = axios.create({
    baseURL: baseURL,
    headers: {
        'Content-Type': 'application/json',
    },
});

interface BackendResponse<T> {
  status: string;
  data: T;
}

const get = async <T>(url: string, params: Record<string, any> = {}, config: AxiosRequestConfig = {}): Promise<BackendResponse<T>> => {
  try {
    const response = await axiosWrapper.get<BackendResponse<T>>(url, { params, ...config });
    return response.data;
  } catch (error) {
    throw error;
  }
};

const post = async <T>(url: string, data: any = {}, config: AxiosRequestConfig = {}): Promise<BackendResponse<T>> => {
  try {
    const response = await axiosWrapper.post<BackendResponse<T>>(url, data, { ...config });
    return response.data;
  } catch (error) {
    throw error;
  }
};

export { axiosWrapper, get, post };
