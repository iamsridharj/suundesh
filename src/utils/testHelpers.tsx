// src/utils/testHelpers.tsx
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { useTheme } from "src/hooks/ui/useTheme";

const RenderWithTheme = ({ children }: { children: React.ReactNode }) => {
  const theme = useTheme();

  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
};

const render = (ui: React.ReactElement, options = {}) => {
  return rtlRender(
    <RenderWithTheme>
      {ui}
    </RenderWithTheme>,
    options
  );
};

export * from '@testing-library/react';
export { render };
