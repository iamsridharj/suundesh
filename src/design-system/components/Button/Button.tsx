import React, { PropsWithChildren } from "react";

import { ButtonProps } from "./Button.types";
import { StyledButton } from "./Button.styled";

export const Button: React.FC<PropsWithChildren<ButtonProps>> = ({ ariaLabel, children, iconPosition, icon, onClick }) => {
    return (
        <StyledButton
            role="button"
            type="button"
            tabIndex={0}
            aria-label={ariaLabel || "btn"}
            onClick={onClick}
            $iconPosition={iconPosition}
        >
            {children}
            {icon && (icon)}
        </StyledButton>
    )
};
