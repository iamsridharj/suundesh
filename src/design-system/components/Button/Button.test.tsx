import { render, screen } from 'src/utils/testHelpers';
import { Button } from './Button';
import '@testing-library/jest-dom';

describe('Button', () => {
    it('renders the button with correct aria-label and tabIndex', () => {
        render(<Button ariaLabel="test-label">Click me</Button>);
        const button = screen.getByRole('button');
        expect(button).toHaveAttribute('type', 'button');
        expect(button).toHaveAttribute('tabIndex', '0');
        expect(button).toHaveAttribute('aria-label', 'test-label');
    });

    it('renders default aria-label when none is provided', () => {
        render(<Button>Click me</Button>);
        const button = screen.getByRole('button');
        expect(button).toHaveAttribute('aria-label', 'btn');
    });

    it('includes the icon when provided', () => {
        const Icon = () => <span>Icon</span>;
        render(<Button icon={<Icon />}>Click me</Button>);
        const button = screen.getByRole('button');
        const icon = screen.getByText('Icon');
        expect(button).toContainElement(icon);
    });

    it('positions the icon correctly when iconPosition is specified', () => {
        const Icon = () => <span>Icon</span>;
        render(<Button icon={<Icon />} iconPosition="left">Click me</Button>);
        const button = screen.getByRole('button');
        expect(button).toHaveStyle(`flex-direction: row-reverse`);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Button>Click me</Button>);
        expect(asFragment()).toMatchSnapshot();
    });
});
