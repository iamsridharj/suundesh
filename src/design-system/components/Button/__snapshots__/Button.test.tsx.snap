// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="btn"
    class="sc-beySPh csGrps"
    role="button"
    tabindex="0"
    type="button"
  >
    Click me
  </button>
</DocumentFragment>
`;
